김현정/week16/back_tracking/쿼드압축 후 개수 세기.java

```diff
@@ -0,0 +1,56 @@
+package back_tracking;
+
+/**
+ * Author    : 김현정
+ * Date      : 2024.09.05(목)
+ * Runtime   :
+ * Memory    :
+ * Algorithm : [programmers]백트래킹 - LV2_68936 쿼드 압축 후 세기
+ */
+class Solution {
+    public static void main(String[] args) {
+        Solution solution = new Solution();
+        int[] answer = solution.solution(new int[][]{{1,1,0,0},{1,0,0,0},{1,0,0,1},{1,1,1,1}});
+        for(int a : answer){
+            System.out.println(a);
+        }
+    }
+
+    static int[] answer;
+    public int[] solution(int[][] arr) {
+        //전체 압축이 가능한지 볼 것
+        //안되면 반으로 줄일 것
+        answer = new int[2];
+        dfs(arr, 0, 0, arr.length);
+
+        return answer;
+    }
+
+    public void dfs(int[][] arr, int x, int y, int size) {
+        boolean flag = true;
+
+        for(int i = x; i < x+ size; i++) {
+            for(int j = y; j < y + size; j++) {
+                if(arr[i][j] != arr[x][y]) {
+                    flag = false;
+                    break;
+                }
+            }
+            if(!flag) break;
+        }
+
+        int newSize = size/2;
+        if(flag) {
+            if(arr[x][y] == 1){
+                answer[1]++;
+            } else {
+                answer[0]++;
+            }
```

Comment:
```suggestion
            answer[arr[x][y]]++;
```

이렇게도 작성될 수 있겠네요.
arr[x][y] 가 명시적이지 않다면 변수에 담에서 해결해도 괜찮을 것 같네요.

Reply:
오 이게 더 보기가 좋네요! 👍